tests/pkg_tweetnacl/main.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       tweetnacl NaCl crypto library example
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <tweetnacl.h>
+#include "ps.h"
+
+static const char message[] = "0123456789abcdef";
+
+#define MLEN (sizeof(message) + crypto_box_ZEROBYTES)
+
+static unsigned char alice_pk[crypto_box_PUBLICKEYBYTES];
+static unsigned char alice_sk[crypto_box_SECRETKEYBYTES];
+static unsigned char bob_pk[crypto_box_PUBLICKEYBYTES];
+static unsigned char bob_sk[crypto_box_SECRETKEYBYTES];
+static unsigned char m[MLEN];
+static unsigned char c[MLEN];
+static const unsigned char n[crypto_box_NONCEBYTES];
+
+static unsigned char result[MLEN];
+
+int main(void)
+{
+    int res;
+
+    printf("Creating keypair ALICE...\n");
+    crypto_box_keypair(alice_pk, alice_sk);
```

Comment:
Uhm... I'm trying to understand what kind of unittests I should write, and what we gain with it.
Actually, to be honest, I think cover with unittests, or even a "normal" test, on cryptographic code is kind of hard task. The test here is just to provide some insights on how to use the library (actually for the use I make of it I used the functions described on this test), and I don't think it aims to actually test the library.

On the other hand, the code is completely independent of RIOT code, except for the random functions which use the module random on RIOT. The code used for the random function is also independent of the library.

In summary, I think any change on the code on the RIOT side will not affect at all this test (e.g. make it fail), so the only source of failure can be upstream changes, which I would suppose should be tested by the upstream CI (if any).

Maybe I'm misunderstanding the purpose of unittests? I see some unittests that cover some crypto libraries, but these libraries are included in the internal code of RIOT, so I see the impact.